panic: runtime error: invalid memory address or nil pointer dereference

I was playing around with the example code given on https://github.com/lovoo/goka/, I wanted to see what would happen with a bunch of separate instances running in parallel, each emitting a message once per second.  Looking over the code it appeared as though what I needed to do was add in a bit of code to make sure each instance had its own LevelDB path, and then add in the bits to emit more than one message:

```
$ diff -u example.go.orig example.go
--- example.go.orig	2017-10-27 10:58:53.000000000 -0700
+++ example.go	2017-10-27 10:36:42.000000000 -0700
@@ -1,11 +1,13 @@
 package main
 
 import (
-	"fmt"
+	"io/ioutil"
 	"log"
 	"os"
 	"os/signal"
+	"path/filepath"
 	"syscall"
+	"time"
 
 	"github.com/lovoo/goka"
 	"github.com/lovoo/goka/codec"
@@ -18,17 +20,20 @@
 )
 
 // emits a single message and leave
-func runEmitter() {
+func runEmitter(n int) {
 	emitter, err := goka.NewEmitter(brokers, topic, new(codec.String))
 	if err != nil {
 		log.Fatalf("error creating emitter: %v", err)
 	}
 	defer emitter.Finish()
-	err = emitter.EmitSync("some-key", "some-value")
-	if err != nil {
-		log.Fatalf("error emitting message: %v", err)
+	for i := 0; i < n; i++ {
+		err = emitter.EmitSync("some-key", "some-value")
+		if err != nil {
+			log.Fatalf("error emitting message: %v", err)
+		}
+		log.Printf("emitted message #%d", i+1)
+		time.Sleep(time.Second)
 	}
-	fmt.Println("message emitted")
 }
 
 // process messages until ctrl-c is pressed
@@ -56,7 +61,15 @@
 		goka.Persist(new(codec.Int64)),
 	)
 
-	p, err := goka.NewProcessor(brokers, g)
+	tdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		log.Fatalf("error creating temp db dir: %v", err)
+	}
+	defer os.RemoveAll(tdir)
+
+	sb := goka.WithStorageBuilder(goka.DefaultStorageBuilder(filepath.Join(tdir, "group")))
+
+	p, err := goka.NewProcessor(brokers, g, sb)
 	if err != nil {
 		log.Fatalf("error creating processor: %v", err)
 	}
@@ -73,6 +86,6 @@
 }
 
 func main() {
-	runEmitter()   // emits one message and stops
-	runProcessor() // press ctrl-c to stop
+	go runEmitter(100) // emits 100 message and stops
+	runProcessor()     // press ctrl-c to stop
 }
```

What I ended up with was: https://play.golang.org/p/vFakftSgcs

Running 6 instances in parallel did what I'd have expected, I could see the example-group-state topic receive a flow of steadily incrementing values until it reached 600 (6 x 100).

When I Ctrl-C to quit, five of them quit gracefully but one of them gave me this:

```
2017/10/27 10:54:52 emitted message #98
2017/10/27 10:54:53 emitted message #99
2017/10/27 10:54:54 emitted message #100
2017/10/27 10:55:02 Processor: rebalancing: map[]
2017/10/27 10:55:02 Processor: rebalancing: map[]
^C2017/10/27 10:55:04 Processor: stopping
2017/10/27 10:55:04 Processor: wait for main goroutine
2017/10/27 10:55:04 Processor: stopped
2017/10/27 10:55:04 Processor: removing partitions
2017/10/27 10:55:04 Processor: shutdown complete
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x12782b0]

goroutine 66 [running]:
github.com/lovoo/goka/kafka.(*producer).run(0xc4200caa20)
	/Users/jimr/Library/gocode/src/github.com/lovoo/goka/kafka/producer.go:78 +0x2e0
created by github.com/lovoo/goka/kafka.NewProducer
	/Users/jimr/Library/gocode/src/github.com/lovoo/goka/kafka/producer.go:39 +0x243
```

--- a/kafka/producer.go
+++ b/kafka/producer.go
@@ -43,9 +43,8 @@ func NewProducer(brokers []string, config *sarama.Config, registry metrics.Regis
 
 func (p *producer) Close() error {
 	close(p.stop)
-	err := p.producer.Close()
 	<-p.done
-	return err
+	return p.producer.Close()
 }
 
 func (p *producer) Emit(topic string, key string, value []byte) *Promise {

